.github/workflows/kuksa_databroker-cli_build.yml

```diff
@@ -148,23 +148,31 @@ jobs:
     - name: "Archiving RISCV64 artifacts"
       uses: actions/upload-artifact@v3
       with:
-        name: databroker--cli-riscv64
+        name: databroker-cli-riscv64
         path: ${{github.workspace}}/dist/riscv64
 
 
   bom:
-    name: Bill of Material Check
+    name: License Compliance Check
     runs-on: ubuntu-latest
-    needs: build-container
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: "Createbom: License check and Dash output generation"
-        working-directory: ${{github.workspace}}/kuksa_databroker/createbom
+      # Follows the pattern from
+      # https://github.com/eclipse/dash-licenses?tab=readme-ov-file#example-rustcargo
+      - name: "Using cargo to create Dash input"
+        working-directory: ${{github.workspace}}/kuksa_databroker/
+        # target all is not really needed, and will also return i.e. wasm deps, however
+        # better safe than sorry, the alternative would be running this for each currently
+        # built target  and combining the lists, but that would need adapting, when
+        # adding targets, or also when i.e. switching between MUSL/glibc. So this is safer
         run: |
-         cargo install cargo-license
-         python3 createbom.py --dash ${{github.workspace}}/dash-databroker-deps ../databroker
+         cargo tree -e normal --prefix none --no-dedupe --target all --all-features  > ${{github.workspace}}/cargodeps
```

Comment:
Just to make sure that I understand things right - as of now we get the same result for both databroker and databroker-cli, right? I.e. we do not try to be smart and just include what is actually used?

Reply:
Yes it does. 

So I think the difference is, the SBOM (not this snippet) we ship in a container, tries to capture more or less precisely what is in the container, i.e. we only copy databroker bin dependencies, i.e. in the buildall shell script 

```
cp ./databroker/databroker_bin.cdx.json ../dist/$target_docker/sbom.json
```
The BOM creation before does not really care "what" is in the BOM, it collect the fact it sees

The DASH think is more a "project-level Eclipse thing", where we check if the licenses in our dependencies are compatible, and whether we "believe what is written on the box", as such the "correct way" to do them might be for all dependencies (also python, go etc.) whenever something is merged. I did not want to refactor that here, as we are still splitting things out here anyway. Also even if this finds things that do not belong to cli but in fact are dabtroker only, doesn't change the fact, that we do need Eclipse Dash clearing. Only issue is, we are currently maybe checking some things double. Options

  1.  keep like as is
  2.  refactor to do this only once for all Rust code in the repo
  3. be a bit more narrow in th workflows for databroker and databroker-cli build with cargo tree


I prefer 1 or 3 for now, let me check 3

Reply:
Ok I tried 3. now. Probably should have done this before writing that wall of text :)